support `Oscar.iso_gap_oscar` for Alnuth's `FieldByMatrices`

ThomasBreuer opened the pull request:
The idea is to get a better interface between Oscar and those functions from Alnuth which used to delegate tasks to Pari/GP, see #2220 for the current situation.

Note that the proposed `Oscar.iso_gap_oscar` and the already available `Oscar.iso_gap_oscar` code that deals with algebraic extensions on the GAP side cannot be used to simply replace the `_OscarField` that has been introduced in #2220, because of the difference between `DefiningPolynomial` and `IntegerDefiningPolynomial` (see my comment in the discussion of #2220).

ThomasBreuer committed:
support `Oscar.iso_gap_oscar` for Alnuth's `FieldByMatrices`

thofma commented on src/GAP/iso_gap_oscar.jl:
```diff
@@ -180,6 +183,46 @@ function _iso_gap_oscar_number_field(FG::GapObj)
+
+     # Construct the isomorphic number field in Oscar.
+     cfs = GAPWrap.CoefficientsOfUnivariatePolynomial(pol)
+     R, x = polynomial_ring(QQ, "x")
```
```suggestion
     R, x = polynomial_ring(QQ, "x", cached = false)
```

ThomasBreuer replied:
Why shall we force the creation of a new polynomial ring whenever a new number field in Oscar gets created?

thofma replied:
I am happy if you want to keep it, but why shall we force a useless dictionary lookup whenever a new number field is created?

fingolfin replied:
Do we have a written down policy when to enable the caching and when not? @fieker?

thofma commented on src/GAP/iso_gap_oscar.jl:
```diff
@@ -180,6 +183,46 @@ function _iso_gap_oscar_number_field(FG::GapObj)
+     cfs = GAPWrap.CoefficientsOfUnivariatePolynomial(pol)
+     R, x = polynomial_ring(QQ, "x")
+     polFO = R(Vector{QQFieldElem}(cfs))
+     FO, _ = number_field(polFO, "z")
```
```suggestion
     FO, _ = number_field(polFO, "z", cached = false)
```

ThomasBreuer committed:
address comments

- Call `number_field` with option `cached = false`.
  (The field will be cached in the map created by `iso_gap_oscar`.)
- Use `Hecke.Globals.Qx` for creating number fields.
- Call `polynomial_ring` over other base rings with option `cached = false`.
  (The polynomial ring will be cached in the map created by `iso_gap_oscar`.)

ThomasBreuer committed:
add the `IsoGAPOscar`method for `IsNumberFieldByMatrices`

fingolfin reviewed (approved):
Looks good to me. Unless @thofma has objections, maybe we can merge it later today?
